core/src/main/java/org/geysermc/geyser/registry/mappings/MappingsConfigReader.java

```diff
@@ -57,21 +58,21 @@ public Path[] getCustomMappingsFiles() {
         }
     }
 
-    public boolean ensureMappingsDirectory(Path mappingsDirectory) {
+    public boolean noMappingsDirectoryPresent(Path mappingsDirectory) {
         if (!Files.exists(mappingsDirectory)) {
             try {
                 Files.createDirectories(mappingsDirectory);
-                return true;
+                return false;
             } catch (IOException e) {
                 GeyserImpl.getInstance().getLogger().error("Failed to create mappings directory", e);
-                return false;
+                return true;
             }
         }
-        return true;
+        return false;
     }
```

Comment:
This avoids `!` when using the method but I find the method itself much less readable - I would suggest reverting these changes